coordinator/src/main/java/org/apache/uniffle/coordinator/strategy/storage/AbstractSelectStorageStrategy.java

```diff
@@ -76,6 +92,50 @@ public void readAndWriteHdfsStorage(FileSystem fs, Path testPath,
     }
   }
 
+  @Override
+  public void detectStorage() {
+    uris = Lists.newCopyOnWriteArrayList(remoteStoragePathRankValue.entrySet());
+    if (remoteStoragePathRankValue.size() > 1) {
+      CountDownLatch countDownLatch = new CountDownLatch(uris.size());
+      uris.parallelStream().forEach(uri -> {
+        if (uri.getKey().startsWith(ApplicationManager.getPathSchema().get(0))) {
+          Path remotePath = new Path(uri.getKey());
+          String rssTest = uri.getKey() + "/rssTest-" + getCoordinatorId()
+              + Thread.currentThread().getName();
+          Path testPath = new Path(rssTest);
+          RankValue rankValue = remoteStoragePathRankValue.get(uri.getKey());
+          rankValue.setHealthy(new AtomicBoolean(true));
+          long startWriteTime = System.currentTimeMillis();
+          try {
+            FileSystem fs = HadoopFilesystemProvider.getFilesystem(remotePath, hdfsConf);
+            for (int j = 0; j < readAndWriteTimes; j++) {
+              readAndWriteHdfsStorage(fs, testPath, uri.getKey(), rankValue);
+            }
+          } catch (Exception e) {
+            LOG.error("Storage read and write error, we will not use this remote path {}.", uri, e);
+            rankValue.setHealthy(new AtomicBoolean(false));
+          } finally {
+            if (this instanceof LowestIOSampleCostSelectStorageStrategy) {
+              ((LowestIOSampleCostSelectStorageStrategy) this)
+                  .sortPathByRankValue(uri.getKey(), rssTest, startWriteTime, hdfsConf);
+            } else if (this instanceof AppBalanceSelectStorageStrategy) {
+              ((AppBalanceSelectStorageStrategy) this)
+                  .sortPathByRankValue(uri.getKey(), rssTest, hdfsConf);
+            } else {
+              LOG.error("Failed to sort path by detectStorage!");
+            }
+            countDownLatch.countDown();
```

Comment:
It is better to move this line inside the `finally` block.

Reply:
Updated.